src/frontends/pytorch/src/op/embedding.cpp

```diff
@@ -13,10 +14,12 @@ namespace pytorch {
 namespace op {
 
 OutputVector translate_embedding(NodeContext& context) {
+    // aten::embedding(Tensor weight, Tensor indices, SymInt padding_idx=-1, bool scale_grad_by_freq=False, bool
+    // sparse=False) parameters 2, 3 and 4 used only for training case and connected with gradients propagation
```

Comment:
I don't think `sparse` is used only for training. `sparse` means that weight is sparse tensor, not dense. Also `padding_idx` is unclear how it is only for training.

Reply:
according to docs:
sparse ([bool](https://docs.python.org/3/library/functions.html#bool), optional) – If True, gradient w.r.t. weight matrix will be a sparse tensor. See Notes for more details regarding sparse gradients.



Note

Keep in mind that only a limited number of optimizers support sparse gradients: currently it’s optim.SGD (CUDA and CPU), optim.SparseAdam (CUDA and CPU) and optim.Adagrad (CPU)

https://pytorch.org/docs/stable/generated/torch.nn.Embedding.html#torch.nn.Embedding

Reply:
padding_idx ([int](https://docs.python.org/3/library/functions.html#int), optional) – If specified, the entries at padding_idx do not contribute to the gradient; therefore, the embedding vector at padding_idx is not updated during training, i.e. it remains as a fixed “pad”. For a newly constructed Embedding, the embedding vector at padding_idx will default to all zeros, but can be updated to another value to be used as the padding vector.

so it is padding for gradients.

Reply:
other parameters like max_norm and norm_type made on weights, made before aten::embeddings, they are not a part of op

Reply:
Okay, thank you for this analysis, that would be great if you add more detailed comment about why we skip those inputs.

Reply:
updated